sdk/storage/storage-queue/test/utils/recorder.ts

```diff
@@ -0,0 +1,417 @@
+      importNock + "\n" + "module.exports.testInfo = " + JSON.stringify(this.uniqueTestInfo) + "\n"
+    );
+
+    const accountName = env.ACCOUNT_NAME as string;
+    for (const fixture of fixtures) {
+      // '://' added so we can be sure that the substitutions happen only in urls
+      // We're not matching query string parameters because they may contain sensitive information, and Nock does not allow us to customize it easily
+      const updatedFixture = fixture
+        .replace(new RegExp("://" + accountName, "g"), "://fakestorageaccount")
+        .replace(/\.query\(.*\)/, ".query(true)");
+      file.write(filterSecrets(updatedFixture) + "\n");
+    }
+
+    file.end();
+
+    nock.recorder.clear();
+    nock.restore();
+  }
+}
+
+// To better understand how this class works, it's necessary to comprehend how HTTP async requests are made:
+// A new request object is created
+//    let req = new XMLHttpRequest();
+// The request is opened with some important information
+//    req.open(method, url, async, user, password);
+// Since we're dealing with an async request, we must set a way to know when the response is ready
+//    req.onreadystatechange = function() {
+//      if (req.readyState === 4) do_something;
+//    }
+// Finally, the request is sent to the server
+//    req.send(data);
+
+// Nise module does not have a native implementation of record/playback like Nock does
+// This class overrides requests' 'open', 'send' and 'onreadystatechange' functions, adding our own code to them to deal with requests
+class NiseRecorder extends Recorder {
+  private readonly sasQueryParameters = ["se", "sig", "sp", "spr", "srt", "ss", "st", "sv"];
+  private recordings: any[] = [];
+
+  constructor(testHierarchy: string, testTitle: string) {
+    super("browsers", testHierarchy, testTitle, "json");
+  }
+
+  // Inserts a request/response pair into the recordings array
+  private async recordRequest(request: any, data: any): Promise<void> {
+    const responseHeaders: any = {};
+    const responseHeadersPairs = request.getAllResponseHeaders().split("\r\n");
+    for (const pair of responseHeadersPairs) {
+      const [key, value] = pair.split(": ");
+      responseHeaders[key] = value;
+    }
+
+    // We're not storing SAS Query Parameters because they may contain sensitive information
+    // We're ignoring the "_" parameter as well because it's not being added by our code
+    // More info on "_": https://stackoverflow.com/questions/3687729/who-add-single-underscore-query-parameter
+    const parsedUrl = queryString.parseUrl(request.url);
+    const query: any = {};
+    for (const param in parsedUrl.query) {
+      if (!this.sasQueryParameters.includes(param) && param !== "_") {
+        query[param] = parsedUrl.query[param];
+      }
+    }
+
+    const accountName = env.ACCOUNT_NAME;
+    this.recordings.push({
+      method: request.method,
+      url: parsedUrl.url.replace(new RegExp(accountName, "g"), "fakestorageaccount"),
```

Comment:
We can just keep it as `url: parsedUrl.url`, given that we'll replace it in the filter later. Remember to remove the `accountName` declaration as well.